src/wallet/wallet.cpp

```diff
@@ -1037,6 +1037,25 @@ bool CWallet::FindNotesDataAndAddMissingIVKToKeystore(const CTransaction& tx, Op
     return true;
 }
 
+void CWallet::AddExternalNotesDataToTx(CWalletTx& wtx) const
+{
+    if (HasSaplingSPKM() && wtx.IsShieldedTx()) {
+        const uint256& txId = wtx.GetHash();
+        // Add the external outputs.
+        SaplingOutPoint op {txId, 0};
+        for (unsigned int i = 0; i < wtx.sapData->vShieldedOutput.size(); i++) {
+            op.n = i;
+            if (wtx.mapSaplingNoteData.count(op)) continue;     // internal output
```

Comment:
This could be for an internal output or because of this method being called twice. `AddToWalletIfInvolvingMe` isn't guarding it.

Reply:
sure. in any case we want to skip and not re-decrypt again...